src/rgw/rgw_rest_swift.cc

```diff
@@ -1404,6 +1406,83 @@ RGWOp *RGWHandler_REST_Service_SWIFT::op_delete()
   return NULL;
 }
 
+static bool is_web_mode(const req_state * const s) {
+  const boost::string_ref webmode = s->info.env->get("HTTP_X_WEB_MODE", "");
+  return webmode == "true";
+}
+
+static bool is_website_req(const req_state * const s)
+{
+  /* Static website works only with the GET or HEAD method. Nothing more. */
+  const std::set<boost::string_ref> ws_methods = { "GET", "HEAD" };
+  if (ws_methods.count(s->info.method) == 0) {
+    return false;
+  }
+
+  /* We also need to handle early failures from the authentication system.
+   * In such cases req_state::auth_identity may be empty. Website treats them
+   * the same way as anonymous. */
+  if (! s->auth_identity) {
+    return true;
+  }
+
+  /* Swift serves websites only for anonymous requests unless client explicitly
+   * requested this behaviour by supplying X-Web-Mode HTTP header set to true. */
+  if (s->auth_identity->is_anonymous() || is_web_mode(s)) {
+    return true;
+  }
+
+  return false;
+}
+
+RGWOp* RGWHandler_REST_Bucket_SWIFT::get_ws_index_op()
+{
+  // retarget to get obj
+  s->object = s->bucket_info.website_conf.get_swift_index_doc();
+
+  auto getop = new RGWGetObj_ObjStore_SWIFT;
+  getop->set_get_data(true);
+
+  return getop;
+}
+
+RGWOp* RGWHandler_REST_Bucket_SWIFT::get_ws_listing_op()
+{
+  return nullptr;
+}
+
+int RGWHandler_REST_Bucket_SWIFT::retarget(RGWOp* op, RGWOp** new_op)
+{
+  ldout(s->cct, 10) << "Starting retarget" << dendl;
+  RGWOp* op_override = nullptr;
+
+  /* In Swift static web content is served if the request is anonymous or
+   * has W-Web-Mode HTTP header specified to true. */
+  if (is_website_req(s)) {
```

Comment:
In Swift the static web has a global range. If the middleware is enabled, similar checks are always executed. There are no black/white lists for containers nor a dedicated endpoint concepts. On the other hand, the checks are quietly low-weighted - they are basing only on headers or already loaded data.